### PR TITLE
fix: allow the sounds to play on windows

### LINKED ISSUE #1: Makes vscode unusable
**Looking forward for reporting a security issue:**
Please report security issues by following our security policy: `https://contribute.freecodecamp.org/#/security`

**Describe the bug**
Every time I type a letter, it plays this strange sound effect like something suddenly starting and stopping. It is very distracting. 

Plus it did more or less bring my visual studio code to a crawl. 

**To Reproduce**
1. Install the extension 
2. Activate Campfire mode
3. Open a new window and type

**Expected behavior**
Vscode should be allowing me to edit and there should be some lovely guitar strums in the background. 

**Additional context**
I am using Windows if that is helpful. While I am not super familiar with vscode extension development, I am pretty sure the sounds wrong are be loaded wrong.

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #1

There was an attempt at getting the audio found by using the Visual Studio Extension context, but it proceeds to give errors.

### CODE REVIEW

**Comment by a2937** [comment]
The way I kept building my local development version of the plugin, everything ended up in the out folder funnily enough.  And on the actual extension itself, it does the same thing (and includes a nested audio folder). 

image

**Comment by a2937** [comment]
Okay so it seems like one of the problems I'm having is that it needs a commandline audio player installed on your computer. Or to try to have a web view so it can play sounds. Personally I'm leaning towards a web view. 

At least I know Windows has Powershell built in. 

Edit: A web view might be way too much for something like this.

File: package.json [hunk 1]
```diff
@@ -42,7 +44,7 @@
     "@vscode/test-cli": "0.0.9",
     "@vscode/test-electron": "2.4.0",
     "eslint": "9.8.0",
-    "typescript": "5.4.5"
+    "typescript": "^5.5.4"
```
Comment (naomi-lgbt) [inline package.json]: I'm fine with updating the Typescript version, but it needs to be pinned.

File: src/sound.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
+
 /**
  * @copyright freeCodeCamp
  * @license BSD-3
  * @author Naomi Carrigan
  */
-
```
Comment (naomi-lgbt) [inline src/sound.ts]: This formatting change should be reverted

File: src/sound.ts [hunk 3]
```diff
@@ -12,17 +12,29 @@ import type vscode from "vscode";
...
+): void => {
   const audio = player();
+
   const selected
-  = urls[Math.floor(Math.random() * urls.length)] ?? "Guitar-Sample-01.mp3";
+    = urls[Math.floor(Math.random() * urls.length)] ?? "Guitar-Sample-01.mp3";
+
+  if (process.platform === "win32") {
+    audio.player = "powershell";
+  }
```
Comment (naomi-lgbt) [inline src/sound.ts]: We should not need to manually set the player to use - `powershell` is [included in the package's default detections](https://www.npmjs.com/package/play-sound#options)

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: src/sound.ts (hunk 3 above)
Comment (a2937) [reply src/sound.ts]: I'm confused. How does it do detection?

Comment (a2937) [reply src/sound.ts]: I thought it was manual?

File: package.json (hunk 1 above)
Comment (a2937) [reply package.json]: That's fine with me.

File: src/sound.ts (hunk 2 above)
Comment (a2937) [reply src/sound.ts]: No problem artichoke.

File: src/sound.ts (hunk 3 above)
Comment (naomi-lgbt) [reply src/sound.ts]: https://github.com/shime/play-sound/blob/25989f8bcd71665456176174d70c8c75f55bad81/index.js#L21

Comment (a2937) [reply src/sound.ts:27]: Naomi can you humor me and tell me what Error Code 1 means when you're trying to play an audio file on Windows?

Comment (a2937) [reply src/sound.ts]: Is there a way to test to see if my changes actually worked? 

Other than manual I mean.

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**